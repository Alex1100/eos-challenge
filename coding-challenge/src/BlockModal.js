import React from 'react';
import { Modal } from 'react-materialize';

const BlockModal = (props) => {
  const block = props.block;
  const index = props.index;

  return (
    <div key={`block_${block.block_num + index}`}>
      <Modal
        header={`block #${block.block_num}`}
        trigger={
          <div className="liveStreamBlock">
            <p>
              <span className="text-gold text-wrap">
                Block Merkle Root
              </span>
            </p>
            <p>
              <span className="text-white">
                {block.block_mroot}
              </span>
            </p>
            <p>
              <span className="text-gold">
                Timestamp
              </span>
            </p>
            <p>
              <span className="text-white">
                {block.timestamp}
              </span>
            </p>
            <p>
              <span className="text-gold">
                Input Transactions
              </span>
            </p>
            <p>
              <span className="text-white">
                {block.input_transactions}
              </span>
            </p>
          </div>
        }>
        <div className="blockContents text-wrap">
          {JSON.stringify(block, null, 2)}
        </div>
      </Modal>
    </div>
  );
};

export default BlockModal;
