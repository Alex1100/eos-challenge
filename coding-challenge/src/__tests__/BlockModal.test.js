const requestAnimationFrame = global.requestAnimationFrame = callback => {
  setTimeout(callback, 0);
}

import React from 'react';
import { shallow } from 'enzyme';
import BlockModal from '../BlockModal';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter(), disableLifecycleMethods: true });

const props = {
  "__typename": "Block",
  "action_mroot": "460911ed2fcfe387eae38d7fa00c2b1c2b09caaa0292281d6a66aa891076ed41",
  "block_mroot": "6c6afdb6f5a80d1461a6f5c964da7163ee7b851d8b6d53592110003653420ae9",
  "block_num": 8425,
  "error": "",
  "input_transactions": 0,
  "previous": "000020e838ed27800e8b8618de018ab1936a63d1b973eabd44cbb836528aff25",
  "producer": "eosio",
  "schedule_version": 0,
  "timestamp": "2018-04-25T19:15:18.500",
  "transaction_mroot": "d6a18ad9a017ef31376289aeceaf41369abc1ab3291a949e57d52295a219a580",
};

describe("BlockModal", () => {
  const BlockModalComponent = shallow(<BlockModal block={props} index={0} />);

  it('tests to see if the modal renders with the block props', () => {
    expect(BlockModalComponent).toMatchSnapshot();
  });
})
