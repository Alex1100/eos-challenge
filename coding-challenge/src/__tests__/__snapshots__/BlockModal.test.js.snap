// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockModal tests to see if the modal renders with the block props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BlockModal
    block={
      Object {
        "__typename": "Block",
        "action_mroot": "460911ed2fcfe387eae38d7fa00c2b1c2b09caaa0292281d6a66aa891076ed41",
        "block_mroot": "6c6afdb6f5a80d1461a6f5c964da7163ee7b851d8b6d53592110003653420ae9",
        "block_num": 8425,
        "error": "",
        "input_transactions": 0,
        "previous": "000020e838ed27800e8b8618de018ab1936a63d1b973eabd44cbb836528aff25",
        "producer": "eosio",
        "schedule_version": 0,
        "timestamp": "2018-04-25T19:15:18.500",
        "transaction_mroot": "d6a18ad9a017ef31376289aeceaf41369abc1ab3291a949e57d52295a219a580",
      }
    }
    index={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "block_8425",
    "nodeType": "host",
    "props": Object {
      "children": <Modal
        actions={
          Array [
            <Button
              flat={true}
              modal="close"
              node="button"
              waves="light"
            >
              Close
            </Button>,
          ]
        }
        bottomSheet={false}
        fixedFooter={false}
        header="block #8425"
        modalOptions={Object {}}
        trigger={
          <div
            className="liveStreamBlock"
          >
            <p>
              <span
                className="text-gold text-wrap"
              >
                Block Merkle Root
              </span>
            </p>
            <p>
              <span
                className="text-white"
              >
                6c6afdb6f5a80d1461a6f5c964da7163ee7b851d8b6d53592110003653420ae9
              </span>
            </p>
            <p>
              <span
                className="text-gold"
              >
                Timestamp
              </span>
            </p>
            <p>
              <span
                className="text-white"
              >
                2018-04-25T19:15:18.500
              </span>
            </p>
            <p>
              <span
                className="text-gold"
              >
                Input Transactions
              </span>
            </p>
            <p>
              <span
                className="text-white"
              >
                0
              </span>
            </p>
          </div>
        }
      >
        <div
          className="blockContents text-wrap"
        >
          {
  "__typename": "Block",
  "action_mroot": "460911ed2fcfe387eae38d7fa00c2b1c2b09caaa0292281d6a66aa891076ed41",
  "block_mroot": "6c6afdb6f5a80d1461a6f5c964da7163ee7b851d8b6d53592110003653420ae9",
  "block_num": 8425,
  "error": "",
  "input_transactions": 0,
  "previous": "000020e838ed27800e8b8618de018ab1936a63d1b973eabd44cbb836528aff25",
  "producer": "eosio",
  "schedule_version": 0,
  "timestamp": "2018-04-25T19:15:18.500",
  "transaction_mroot": "d6a18ad9a017ef31376289aeceaf41369abc1ab3291a949e57d52295a219a580"
}
        </div>
      </Modal>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "actions": Array [
          <Button
            flat={true}
            modal="close"
            node="button"
            waves="light"
          >
            Close
          </Button>,
        ],
        "bottomSheet": false,
        "children": <div
          className="blockContents text-wrap"
        >
          {
  "__typename": "Block",
  "action_mroot": "460911ed2fcfe387eae38d7fa00c2b1c2b09caaa0292281d6a66aa891076ed41",
  "block_mroot": "6c6afdb6f5a80d1461a6f5c964da7163ee7b851d8b6d53592110003653420ae9",
  "block_num": 8425,
  "error": "",
  "input_transactions": 0,
  "previous": "000020e838ed27800e8b8618de018ab1936a63d1b973eabd44cbb836528aff25",
  "producer": "eosio",
  "schedule_version": 0,
  "timestamp": "2018-04-25T19:15:18.500",
  "transaction_mroot": "d6a18ad9a017ef31376289aeceaf41369abc1ab3291a949e57d52295a219a580"
}
        </div>,
        "fixedFooter": false,
        "header": "block #8425",
        "modalOptions": Object {},
        "trigger": <div
          className="liveStreamBlock"
        >
          <p>
            <span
              className="text-gold text-wrap"
            >
              Block Merkle Root
            </span>
          </p>
          <p>
            <span
              className="text-white"
            >
              6c6afdb6f5a80d1461a6f5c964da7163ee7b851d8b6d53592110003653420ae9
            </span>
          </p>
          <p>
            <span
              className="text-gold"
            >
              Timestamp
            </span>
          </p>
          <p>
            <span
              className="text-white"
            >
              2018-04-25T19:15:18.500
            </span>
          </p>
          <p>
            <span
              className="text-gold"
            >
              Input Transactions
            </span>
          </p>
          <p>
            <span
              className="text-white"
            >
              0
            </span>
          </p>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "{
  \\"__typename\\": \\"Block\\",
  \\"action_mroot\\": \\"460911ed2fcfe387eae38d7fa00c2b1c2b09caaa0292281d6a66aa891076ed41\\",
  \\"block_mroot\\": \\"6c6afdb6f5a80d1461a6f5c964da7163ee7b851d8b6d53592110003653420ae9\\",
  \\"block_num\\": 8425,
  \\"error\\": \\"\\",
  \\"input_transactions\\": 0,
  \\"previous\\": \\"000020e838ed27800e8b8618de018ab1936a63d1b973eabd44cbb836528aff25\\",
  \\"producer\\": \\"eosio\\",
  \\"schedule_version\\": 0,
  \\"timestamp\\": \\"2018-04-25T19:15:18.500\\",
  \\"transaction_mroot\\": \\"d6a18ad9a017ef31376289aeceaf41369abc1ab3291a949e57d52295a219a580\\"
}",
          "className": "blockContents text-wrap",
        },
        "ref": null,
        "rendered": "{
  \\"__typename\\": \\"Block\\",
  \\"action_mroot\\": \\"460911ed2fcfe387eae38d7fa00c2b1c2b09caaa0292281d6a66aa891076ed41\\",
  \\"block_mroot\\": \\"6c6afdb6f5a80d1461a6f5c964da7163ee7b851d8b6d53592110003653420ae9\\",
  \\"block_num\\": 8425,
  \\"error\\": \\"\\",
  \\"input_transactions\\": 0,
  \\"previous\\": \\"000020e838ed27800e8b8618de018ab1936a63d1b973eabd44cbb836528aff25\\",
  \\"producer\\": \\"eosio\\",
  \\"schedule_version\\": 0,
  \\"timestamp\\": \\"2018-04-25T19:15:18.500\\",
  \\"transaction_mroot\\": \\"d6a18ad9a017ef31376289aeceaf41369abc1ab3291a949e57d52295a219a580\\"
}",
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "block_8425",
      "nodeType": "host",
      "props": Object {
        "children": <Modal
          actions={
            Array [
              <Button
                flat={true}
                modal="close"
                node="button"
                waves="light"
              >
                Close
              </Button>,
            ]
          }
          bottomSheet={false}
          fixedFooter={false}
          header="block #8425"
          modalOptions={Object {}}
          trigger={
            <div
              className="liveStreamBlock"
            >
              <p>
                <span
                  className="text-gold text-wrap"
                >
                  Block Merkle Root
                </span>
              </p>
              <p>
                <span
                  className="text-white"
                >
                  6c6afdb6f5a80d1461a6f5c964da7163ee7b851d8b6d53592110003653420ae9
                </span>
              </p>
              <p>
                <span
                  className="text-gold"
                >
                  Timestamp
                </span>
              </p>
              <p>
                <span
                  className="text-white"
                >
                  2018-04-25T19:15:18.500
                </span>
              </p>
              <p>
                <span
                  className="text-gold"
                >
                  Input Transactions
                </span>
              </p>
              <p>
                <span
                  className="text-white"
                >
                  0
                </span>
              </p>
            </div>
          }
        >
          <div
            className="blockContents text-wrap"
          >
            {
  "__typename": "Block",
  "action_mroot": "460911ed2fcfe387eae38d7fa00c2b1c2b09caaa0292281d6a66aa891076ed41",
  "block_mroot": "6c6afdb6f5a80d1461a6f5c964da7163ee7b851d8b6d53592110003653420ae9",
  "block_num": 8425,
  "error": "",
  "input_transactions": 0,
  "previous": "000020e838ed27800e8b8618de018ab1936a63d1b973eabd44cbb836528aff25",
  "producer": "eosio",
  "schedule_version": 0,
  "timestamp": "2018-04-25T19:15:18.500",
  "transaction_mroot": "d6a18ad9a017ef31376289aeceaf41369abc1ab3291a949e57d52295a219a580"
}
          </div>
        </Modal>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "actions": Array [
            <Button
              flat={true}
              modal="close"
              node="button"
              waves="light"
            >
              Close
            </Button>,
          ],
          "bottomSheet": false,
          "children": <div
            className="blockContents text-wrap"
          >
            {
  "__typename": "Block",
  "action_mroot": "460911ed2fcfe387eae38d7fa00c2b1c2b09caaa0292281d6a66aa891076ed41",
  "block_mroot": "6c6afdb6f5a80d1461a6f5c964da7163ee7b851d8b6d53592110003653420ae9",
  "block_num": 8425,
  "error": "",
  "input_transactions": 0,
  "previous": "000020e838ed27800e8b8618de018ab1936a63d1b973eabd44cbb836528aff25",
  "producer": "eosio",
  "schedule_version": 0,
  "timestamp": "2018-04-25T19:15:18.500",
  "transaction_mroot": "d6a18ad9a017ef31376289aeceaf41369abc1ab3291a949e57d52295a219a580"
}
          </div>,
          "fixedFooter": false,
          "header": "block #8425",
          "modalOptions": Object {},
          "trigger": <div
            className="liveStreamBlock"
          >
            <p>
              <span
                className="text-gold text-wrap"
              >
                Block Merkle Root
              </span>
            </p>
            <p>
              <span
                className="text-white"
              >
                6c6afdb6f5a80d1461a6f5c964da7163ee7b851d8b6d53592110003653420ae9
              </span>
            </p>
            <p>
              <span
                className="text-gold"
              >
                Timestamp
              </span>
            </p>
            <p>
              <span
                className="text-white"
              >
                2018-04-25T19:15:18.500
              </span>
            </p>
            <p>
              <span
                className="text-gold"
              >
                Input Transactions
              </span>
            </p>
            <p>
              <span
                className="text-white"
              >
                0
              </span>
            </p>
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "{
  \\"__typename\\": \\"Block\\",
  \\"action_mroot\\": \\"460911ed2fcfe387eae38d7fa00c2b1c2b09caaa0292281d6a66aa891076ed41\\",
  \\"block_mroot\\": \\"6c6afdb6f5a80d1461a6f5c964da7163ee7b851d8b6d53592110003653420ae9\\",
  \\"block_num\\": 8425,
  \\"error\\": \\"\\",
  \\"input_transactions\\": 0,
  \\"previous\\": \\"000020e838ed27800e8b8618de018ab1936a63d1b973eabd44cbb836528aff25\\",
  \\"producer\\": \\"eosio\\",
  \\"schedule_version\\": 0,
  \\"timestamp\\": \\"2018-04-25T19:15:18.500\\",
  \\"transaction_mroot\\": \\"d6a18ad9a017ef31376289aeceaf41369abc1ab3291a949e57d52295a219a580\\"
}",
            "className": "blockContents text-wrap",
          },
          "ref": null,
          "rendered": "{
  \\"__typename\\": \\"Block\\",
  \\"action_mroot\\": \\"460911ed2fcfe387eae38d7fa00c2b1c2b09caaa0292281d6a66aa891076ed41\\",
  \\"block_mroot\\": \\"6c6afdb6f5a80d1461a6f5c964da7163ee7b851d8b6d53592110003653420ae9\\",
  \\"block_num\\": 8425,
  \\"error\\": \\"\\",
  \\"input_transactions\\": 0,
  \\"previous\\": \\"000020e838ed27800e8b8618de018ab1936a63d1b973eabd44cbb836528aff25\\",
  \\"producer\\": \\"eosio\\",
  \\"schedule_version\\": 0,
  \\"timestamp\\": \\"2018-04-25T19:15:18.500\\",
  \\"transaction_mroot\\": \\"d6a18ad9a017ef31376289aeceaf41369abc1ab3291a949e57d52295a219a580\\"
}",
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
